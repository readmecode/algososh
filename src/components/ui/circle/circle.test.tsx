import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";
import { Circle } from "./circle";

describe("test uiCircle", () => {
  test("without letters", () => {
    const tree = renderer.create(<Circle />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with letters", () => {
    const tree = renderer.create(<Circle letter={"H"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with head", () => {
    const tree = renderer.create(<Circle head={"head"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with react-element in head", () => {
    const reactElement = <h1>test header</h1>;
    const tree = renderer.create(<Circle head={reactElement} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with tail", () => {
    const tree = renderer.create(<Circle tail={"tail"} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with react-element in tail", () => {
    const reactElement = <h1>test tail</h1>;
    const tree = renderer.create(<Circle tail={reactElement} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("with props isSmall={true}", () => {
    const tree = renderer.create(<Circle isSmall={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("type of condition default", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Default} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("type of condition changing", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Changing} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("type of condition modified", () => {
    const tree = renderer
      .create(<Circle state={ElementStates.Modified} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
