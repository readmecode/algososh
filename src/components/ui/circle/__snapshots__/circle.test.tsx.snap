// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test uiCircle type of condition changing 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   changing"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle type of condition default 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle type of condition modified 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   modified"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle with head 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circle-head"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle with letters 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    >
      H
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle with props isSmall={true} 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle  small default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle with react-element in head 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  >
    <h1>
      test header
    </h1>
  </div>
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`test uiCircle with react-element in tail 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  >
    <h1>
      test tail
    </h1>
  </div>
</div>
`;

exports[`test uiCircle with tail 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circle-tail"
  >
    tail
  </div>
</div>
`;

exports[`test uiCircle without letters 1`] = `
<div
  className="content "
  data-testid="contetnt"
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    className="circle   default"
    data-testid="circle"
  >
    <p
      className="text text_type_circle text_color_input letter"
      data-testid="circle-letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;
